docs/snippets/ov_preprocessing_migration.cpp

```diff
@@ -0,0 +1,121 @@
+// Copyright (C) 2018-2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+#include <openvino/runtime/core.hpp>
+#include <openvino/opsets/opset8.hpp>
+#include <openvino/core/preprocess/pre_post_process.hpp>
+
+#include "inference_engine.hpp"
+
+int main_new() {
+    std::string model_path;
+
+    ov::Core core;
+    std::shared_ptr<ov::Model> model = core.read_model(model_path);
+    ov::preprocess::PrePostProcessor ppp(model);
+
+    {
+    //! [ov_mean_scale]
+ov::preprocess::PrePostProcessor ppp(model);
+ov::preprocess::InputInfo& input = ppp.input("tensor_name");
+// we only need to know where is C dimension
+input.model().set_layout("???C");
+// specify scale and mean values, order of operations is important
+input.preprocess().mean(0.1f).scale({ 0.5f, 0.05f, 0.01f });
+// insert preprocessing operations to the 'model'
+model = ppp.build();
+    //! [ov_mean_scale]
+    }
+
+    {
+    //! [ov_conversions]
+ov::preprocess::PrePostProcessor ppp(model);
+ov::preprocess::InputInfo& input = ppp.input("tensor_name");
+input.tensor().set_layout("NHWC").set_element_type(ov::element::u8);
+input.model().set_layout("NCHW");
+// layout and precision conversion is inserted automatically,
+// because tensor != model_input formats
+model = ppp.build();
+    //! [ov_conversions]
+    }
+
+    {
+    //! [ov_color_space]
+ov::preprocess::PrePostProcessor ppp(model);
+ov::preprocess::InputInfo& input = ppp.input("tensor_name");
+input.tensor().set_color_format(ov::preprocess::ColorFormat::NV12_TWO_PLANES);
+// add NV12 to BGR conversion
+input.preprocess().convert_color(ov::preprocess::ColorFormat::BGR);
+// and insert operations to the model
+model = ppp.build();
+    //! [ov_color_space]
+    }
+
+    {
+    //! [ov_image_scale]
+ov::preprocess::PrePostProcessor ppp(model);
+ov::preprocess::InputInfo& input = ppp.input("tensor_name");
+// scale from the specified tensor size
+input.tensor().set_spatial_static_shape(448, 448);
```

Comment:
you don't have 448x448 in old snippet. Suggesting to use `set_spatial_dynamic_shape` here, or maybe even `set_shape({1, 3, -1, -1});`

Reply:
I supposed that if user will use dynamic shapes - he can face with the issue that GPU and mane other plugins don't support it. So, decided to keep explicit shape